generator/all_test.go

```diff
@@ -15,11 +15,44 @@
 package main_test
 
 import (
+	"bufio"
 	"errors"
+	"io/fs"
+	"os"
 	"os/exec"
+	"regexp"
+	"strings"
 	"testing"
 )
 
+var googleHeader = regexp.MustCompile(`^// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2\.0 \(the "License"\);`)
+
+func TestHeaders(t *testing.T) {
+	sfs := os.DirFS(".")
+	fs.WalkDir(sfs, ".", func(path string, d fs.DirEntry, _ error) error {
+		if d.IsDir() {
+			if d.Name() == "testdata" {
+				return fs.SkipDir
+			}
+			return nil
+		}
+		if !strings.HasSuffix(path, ".go") {
```

Comment:
It would be nice to have a single program to validate `.go`, `.rs`, `.toml`, `.yaml`, and any other files that require copyright notices.